daemon/state.go

```diff
@@ -134,14 +133,12 @@ func (d *Daemon) regenerateRestoredEndpoints(state *endpointRestoreState) {
 	for _, ep := range state.restored {
 		go func(ep *endpoint.Endpoint, epRegenerated chan<- bool) {
 			ep.Mutex.Lock()
-			scopedLog := log.WithField(logfields.EndpointID, ep.ID)
 
-			state := ep.GetStateLocked()
-			if state == endpoint.StateDisconnected || state == endpoint.StateDisconnecting {
```

Comment:
Why did we need to delete this logic? 
wouldn't this still be true , i.e can we not receive endpoints in regenerateRestoredEndpoints in state
`endpoint.StateDisconnected` or `state == endpoint.StateDisconnecting`?

Reply:
See https://github.com/cilium/cilium/pull/4398#discussion_r193624510